Mesh_3/include/CGAL/Mesh_3/initialize_triangulation_from_labeled_image.h

```diff
@@ -114,28 +127,50 @@ void initialize_triangulation_from_labeled_image(C3T3& c3t3,
                                              image.vy()),
                                   image.vz());
 
-  typedef std::vector<std::pair<Bare_point, std::size_t> > Seeds;
+  using Point_indices = std::tuple<std::size_t, std::size_t, std::size_t>;
+  using Seed = std::pair<Point_indices, std::size_t>;
+  using Seeds = std::vector<Seed>;
+  using Subdomain_index = typename Mesh_domain::Subdomain_index;
+
   Seeds seeds;
   Get_point<Bare_point> get_point(&image);
   std::cout << "Searching for connected components..." << std::endl;
   search_for_connected_components_in_labeled_image(image,
                                                    std::back_inserter(seeds),
                                                    CGAL::Emptyset_iterator(),
                                                    transform,
-                                                   get_point,
                                                    Image_word_type());
   std::cout << "  " << seeds.size() << " components were found." << std::endl;
   std::cout << "Construct initial points..." << std::endl;
-  for(typename Seeds::const_iterator it = seeds.begin(), end = seeds.end();
-      it != end; ++it)
+  for(const Seed seed : seeds)
   {
-    const double radius = double(it->second + 1)* max_v;
+    const std::size_t i = get<0>(seed.first);
+    const std::size_t j = get<1>(seed.first);
+    const std::size_t k = get<2>(seed.first);
+
+    const Bare_point seed_point = get_point(i, j, k);
+    Cell_handle seed_cell = tr.locate(cwp(seed_point));
+
+    const boost::optional<Subdomain_index> seed_label
+      = domain.is_in_domain_object()(seed_point);
+    const boost::optional<Subdomain_index> seed_cell_label
+      = (seed_cell == Cell_handle()) //seed_point is OUTSIDE_AFFINE_HULL
+        ? boost::none
+        : domain.is_in_domain_object()(
+            seed_cell->weighted_circumcenter(tr.geom_traits()));
```

Comment:
Avoid `boost::none` to prepare the switch to C++17 `std::optional`:
```suggestion
    const boost::optional<Subdomain_index> seed_cell_label
      = (seed_cell == Cell_handle()) //seed_point is OUTSIDE_AFFINE_HULL
        ? {} // the default constructed optional is empty
        : domain.is_in_domain_object()(
            seed_cell->weighted_circumcenter(tr.geom_traits()));
```